Bundler/Viewer/Unity/Assets/Scripts/Network/Scripts/NetworkAnchor.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using Persistence;
+
+public class NetworkAnchor : MonoBehaviour
+{
+    /// <summary>
+    /// The game object to enable when the initial network anchor is being loaded.
+    /// </summary>
+    public GameObject LoadingAnchorRoot;
+
+    /// <summary>
+    /// The game object to enable when the initial network anchor has been loaded.
+    /// </summary>
+    public GameObject FoundAnchorRoot;
+
+    /// <summary>
+    /// For this to function, the game object must also have the AnchorPersistence behavior applied.
+    /// </summary>
+    private AnchorPersistence anchorPersistence;
+
+    /// <summary>
+    /// For this to function, there must be a global NetworkAnchorManager.
+    /// </summary>
+    private NetworkAnchorManager networkAnchorManager;
+
+    /// <summary>
+    /// For this to function, the game object must also have a local instance of an NetworkAnchorPlayer.
+    /// </summary>
+    private NetworkAnchorPlayer anchorPlayer;
+
+    private PersistenceEventArgs pendingPersistenceEventArgs = null;
+
+    private void Awake()
+    {
+        if (FoundAnchorRoot != null)
+        {
+            FoundAnchorRoot.SetActive(false);
+        }
+
+        if (LoadingAnchorRoot != null)
+        {
+            LoadingAnchorRoot.SetActive(false);
+        }
+    }
+
+    /// <summary>
+    /// Initialization the Network Anchor. Note, this will search of an Anchor Persistence behavior. If not present,
+    /// then this behavior will not function correctly.
+    /// </summary>
+    private void Start()
+    {
+        anchorPersistence = gameObject.GetComponent<AnchorPersistence>();
+        if (anchorPersistence != null)
+        {
+            anchorPersistence.PersistenceEvent += OnPersistenceEvent;
+        }
+        else
+        {
+            Debug.LogError("[NetworkAnchor] Network Anchor can't function correctly when there isn't an Anchor Persistence behaviour applied.");
+        }
+    }
+
+    /// <summary>
+    /// Update anchor state.
+    /// </summary>
+    private void Update()
+    {
+        InitializeAnchorManagerOnce();
```

Comment:
question: why the "Once" in the initialize method names?

Reply:
Renamed....hopefully a little clearer.    Basically I want to say "once we can initialize anchor manager stuff, only do the initialize stuff once"